packages/drivers/routerlicious-driver/src/documentDeltaConnection.ts

```diff
@@ -107,4 +107,14 @@ export class R11sDocumentDeltaConnection extends DocumentDeltaConnection {
 			url: getUrlForTelemetry(this.url, socketIoPath),
 		};
 	}
+
+	/**
+	 * Disconnect from the websocket
+	 */
+	protected disconnectCore(err: IAnyDriverError): void {
+		if (this.socket.connected && err.message !== "Client closing delta connection") {
+			this.socket.emit("abnormal_disconnect", this.clientId, this.documentId, err.message);
```

Comment:
Lets call this "client_disconnect" instead of abnormal disconnect and then also no need to check the "err.message !== "Client closing delta connection"". If socket is connected and client is disconnecting, you can just pass the err message/errorType or maybe both. Checking strings like this, does not look good.

Reply:
@znewton is this ok if you guys receive "Client closing delta connection" for clean disconnects?

Reply:
As the original proposer for "abnormal_disconnect", I'm fine using "client_disconnect" but we wanted to avoid sending that event to the server for _every_ client disconnect, particularly clean ones, since it seemed to just waste bandwidth (at least today when the only thing the server side wants to know is when the disconnection is due to an error). Renaming is fine, but I'd still only emit it when there's an actual error.

Reply:
changed name to `client_disconnect`